tests/src/Kernel/ViewsLinkFieldsTest.php

```diff
@@ -0,0 +1,149 @@
+        'media',
+        'system',
+        'user',
+        'views',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp(): void {
+        parent::setUp();
+
+        $this->installEntitySchema('file');
+        $this->installEntitySchema('media');
+        $this->installEntitySchema('user');
+        $this->installConfig(['user', 'views', 'collabora_online_test']);
+        $this->installSchema('file', ['file_usage']);
+        // Install user module avoid user 1 permissions bypass.
+        \Drupal::moduleHandler()->loadInclude('user', 'install');
+        user_install();
+
+        // Create two medias to check access with different scopes, 'any' and 'own'.
+        $this->createMediaEntity('document');
+        $this->ownMedia = $this->createMediaEntity('document');;
+    }
+
+    /**
+     * Tests link fields.
+     */
+    public function testLinks(): void {
+        // User without permissions can't see links.
+        $this->doTestLinks([
+            'preview' => [FALSE, FALSE],
+            'edit' => [FALSE, FALSE],
+        ],
+        $this->createUser([]));
+        // User with 'Preview' permission can see preview link.
+        $this->doTestLinks([
+            'preview' => [TRUE, TRUE],
+            'edit' => [FALSE, FALSE],
+        ],
+        $this->createUser([
+        'preview document in collabora'
+        ]));
+        // User with 'Edit any' permission can see edit link.
+        $this->doTestLinks([
+            'preview' => [FALSE, FALSE],
+            'edit' => [TRUE, TRUE],
+        ],
+        $this->createUser([
+            'edit any document in collabora'
+        ]));
+        // User with 'Edit own' permission can see edit link for entities they own.
+        $this->doTestLinks([
+            'preview' => [FALSE, FALSE],
+            'edit' => [FALSE, TRUE],
+        ],
+    $this->createUser([
+            'edit own document in collabora'
+        ]));
+    }
+
+    /**
+     * Tests that links behave as expected.
+     *
+     * @param [] $expected_results
+     *   An associative array of expected results keyed by operation.
+     * @param \Drupal\Core\Session\AccountInterface $account
+     *   The user account to be used to run the test;
+     */
+    protected function doTestLinks(array $expected_results, AccountInterface $account) {
+        $this->setCurrentUser(account: $account);
+        // Set the current user as the owner to check 'edit own' access.
+        $this->ownMedia->setOwnerId($account->id())->save();
+        $view = Views::getView('test_collabora_links');
+        $view->preview();
+
+        $info = [
+        'preview' => [
+            'label' => 'Preview in Collabora',
+            'field_id' => 'collabora_preview',
+            'route' => 'collabora-online.view'
+        ],
+        'edit' => [
+            'label' => 'Edit in Collabora',
+            'field_id' => 'collabora_edit',
+            'route' => 'collabora-online.edit'
+        ],
```

Comment:
```suggestion
            'preview' => [
                'label' => 'Preview in Collabora',
                'field_id' => 'collabora_preview',
                'route' => 'collabora-online.view'
            ],
            'edit' => [
                'label' => 'Edit in Collabora',
                'field_id' => 'collabora_edit',
                'route' => 'collabora-online.edit'
            ],
```
I know it is annoying to keep up with the 4 spaces.
And nice to see you keep deliver the different files with the "correct" indentation depending where we are!

Reply:
Fixed other formatting issues, they are everywhere.